pyiceberg/catalog/__init__.py

```diff
@@ -858,6 +875,7 @@ def _update_and_stage_table(
             enforce_validation=current_table is None,
             metadata_location=current_table.metadata_location if current_table else None,
         )
+        io = self._load_file_io(properties=updated_metadata.properties, location=updated_metadata.location)
```

Comment:
should we revert this change since its unrelated to this PR? is `updated_metadata.location` the same as `new_metadata_location`?